pkg/providers/snow/validator.go

```diff
@@ -13,22 +13,28 @@ const (
 )
 
 type Validator struct {
-	awsClients aws.Clients
+	awsClientMap AwsClientMap
 }
 
 func NewValidator(aws aws.Clients) *Validator {
 	return &Validator{
-		awsClients: aws,
+		awsClientMap: NewAwsClientMap(aws),
 	}
 }
 
-func (v *Validator) validateSshKeyPair(ctx context.Context, m *v1alpha1.SnowMachineConfig) error {
+func NewValidatorFromAwsClientMap(awsClientMap AwsClientMap) *Validator {
```

Comment:
curious about this, is there a different constructor and that's why we add the `FromAwsClientMap` suffix?

Reply:
this is for unit testing so that we don't export `awsClientMap` field in the struct